### PR TITLE
Add Theis Benchmark for ComponentTransport

### PR DESCRIPTION
**Pull request by waltherm**
As titled.

### CODE REVIEW

**Comment by bilke** [comment]
@waltherm There are formatting issues in the prj-file:

```

[2019-05-02T12:33:23.490Z] + git merge-base origin/master HEAD
[2019-05-02T12:33:23.490Z] + git diff --check f5e8fa800bf0ca973890b502ecfb6b7212969d54 HEAD -- . :!*.md :!*.pandoc :!*.asc
[2019-05-02T12:33:23.490Z] Tests/Data/Parabolic/ComponentTransport/HC_Theis/HC_theis.prj:50: trailing whitespace.
[2019-05-02T12:33:23.490Z] +                    <!-- The implementation of the ComponentTransport process 
[2019-05-02T12:33:23.490Z] Tests/Data/Parabolic/ComponentTransport/HC_Theis/HC_theis.prj:51: trailing whitespace.
[2019-05-02T12:33:23.490Z] +                    is without Boussinesq approximations, thus, 'storage' is not 
[2019-05-02T12:33:23.490Z] Tests/Data/Parabolic/ComponentTransport/HC_Theis/HC_theis.prj:185: space before tab in indent.
[2019-05-02T12:33:23.490Z] +    	<parameter>
[2019-05-02T12:33:23.490Z] Tests/Data/Parabolic/ComponentTransport/HC_Theis/HC_theis.prj:200: trailing whitespace.
[2019-05-02T12:33:23.490Z] +        
```

**Comment by waltherm** [comment]
@bilke  Hope, I could fix the formatting issues.

**Comment by endJunction** [comment]
@waltherm 

1. In the proposed test the concentration evolution is not tested. Is it intended?
2. The test is axisymmetric, but a 3D test is set up. I think the 2D case is easier to set up.

I moved some files around for consistency, and also reformatted the project file, which contained a mix of tabs and spaces.

**Comment by waltherm** [comment]
> @waltherm
> 
>     1. In the proposed test the concentration evolution is not tested. Is it intended?

Yes, this is intended. I wanted to test the hydraulic part of the HC process, as I have no idea whether this is the same as the LIQUID_FLOW process. (The constant implementation of new process combinations should be discussed at another instance, as this is basically inventing the wheel new every time.)

> 
>     2. The test is axisymmetric, but a 3D test is set up. I think the 2D case is easier to set up.

Yes and no. The test is indeed 2D axisymmetric as the analytical solution is, too, but I want to see the process working in 3D. Testing the process in 2D axisym. does not assure that it works in 3D, so I would like to keep it 3D. (I cannot remember using a 2D axisymmetric setup for a real application.)

> 
> 
> I moved some files around for consistency, and also reformatted the project file, which contained a mix of tabs and spaces.

**Comment by endJunction** [comment]
@waltherm Thanks.

Somebody with Theis test knowledge please have a look. Then it can be merged.

File: web/content/docs/benchmarks/hydro-component/theis/HC_Theis.pandoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+The homogeneous, isotropic domain is defined for the radius $1 < r < 100 m$ and a thickness $b = 10 m$. Saturated intrinsic permeability is $\kappa = 7.6453E-13 m^2$ yielding a transmissivity of $T = 7.5E-05 m^2/s$; porosity is $\phi = 0.2$; specific storage is $S_s = 1.0E-03$ and defined through compressibility $\gamma = 5.0968E-08 s^2/m/kg$ (input tag fluid_density_pressure_difference_ratio is $\frac{\partial \rho}{\partial p} = \frac{S_s}{b \phi g \rho}$ with gravitational acceleration $g = 9.81 m^2/s$).
```
Comment (jbathmann) [inline web/content/docs/benchmarks/hydro-component/theis/HC_Theis.pandoc]: gamma = input tag fluid_density_pressure_difference_ratio = $ \frac{1}{\rho_0} \frac{\partial \rho}{\partial p}

File: web/content/docs/benchmarks/hydro-component/theis/HC_Theis.pandoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+Here, we verify pumping abstraction with Theis for the `ComponentTransport` process using a 3D setup. The setup is adapted from Walther (2014), section 3.2.
+
+
+### Model setup
+
+The setup comprises a 1/8th slice of a full circle (see figure 1). 
+
+{{< img src="../BCs.png" title="Mesh and boundary conditions (BC); blue = outer Dirichlet pressure and concentration BC, red = inner Neumann abstraction BC.">}}
+
+The outer boundary condition is set as Dirichlet with a hydrostatic pressure along the shell surface of the slice equivalent to a head of $h = 0 m$ (i.e. water level equals top of domain). For mass transport, a Dirichlet boundary conditions with concentration $c = 0$ is set at the outer shell. The inner boundary condition is equivalent to the eighth of a total abstraction rate of $Q_t = 15 m^3/d$ for a full cylinder. *NB: In the `ComponentTransport` process, the Neumann BC is given as mass flux and has to be calculated per area, such that the value for the project file is $Q = Q_t / 8 / A \cdot \rho = 2.83542E-03 m^3/s/m^2 \cdot kg/m^3$ with fluid density $\rho = 1000 kg/m^3$ and abstraction area $A = 7.65 m^2$.*
```
Comment (jbathmann) [inline web/content/docs/benchmarks/hydro-component/theis/HC_Theis.pandoc]: The units within this section are a bit hard to read. Maybe one could use m^3/s/m^2 \cdot kg/m^3 = \frac{kg}{s m^2} which directly hints at mass flux.

File: web/content/docs/benchmarks/hydro-component/theis/HC_Theis.pandoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+Here, we verify pumping abstraction with Theis for the `ComponentTransport` process using a 3D setup. The setup is adapted from Walther (2014), section 3.2.
+
+
+### Model setup
+
+The setup comprises a 1/8th slice of a full circle (see figure 1). 
+
+{{< img src="../BCs.png" title="Mesh and boundary conditions (BC); blue = outer Dirichlet pressure and concentration BC, red = inner Neumann abstraction BC.">}}
+
+The outer boundary condition is set as Dirichlet with a hydrostatic pressure along the shell surface of the slice equivalent to a head of $h = 0 m$ (i.e. water level equals top of domain). For mass transport, a Dirichlet boundary conditions with concentration $c = 0$ is set at the outer shell. The inner boundary condition is equivalent to the eighth of a total abstraction rate of $Q_t = 15 m^3/d$ for a full cylinder. *NB: In the `ComponentTransport` process, the Neumann BC is given as mass flux and has to be calculated per area, such that the value for the project file is $Q = Q_t / 8 / A \cdot \rho = 2.83542E-03 m^3/s/m^2 \cdot kg/m^3$ (units equal $\frac{kg}{s m^2}$) with fluid density $\rho = 1000 kg/m^3$ and abstraction area $A = 7.65 m^2$.*
```
Comment (jbathmann) [inline web/content/docs/benchmarks/hydro-component/theis/HC_Theis.pandoc]: fluid density $\rho = 1000 kg/m^3$
-->
 reference fluid density $\rho_0 = 1000 kg/m^3$

File: web/content/docs/benchmarks/hydro-component/theis/HC_Theis.pandoc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+The homogeneous, isotropic domain is defined for the radius $1 < r < 100 m$ and a thickness $b = 10 m$. Saturated intrinsic permeability is $\kappa = 7.6453E-13 m^2$ yielding a transmissivity of $T = 7.5E-05 m^2/s$; porosity is $\phi = 0.2$; specific storage is $S_s = 1.0E-03$ and defined through compressibility $\gamma = 5.0968E-08 s^2/m/kg$ (input tag fluid_density_pressure_difference_ratio is $\gamma = \frac{1}{\rho_0} \frac{\partial \rho}{\partial p}$, which can be used to incorporate $S_s$ with $\gamma = \frac{S_s}{b \phi g \rho}$ with gravitational acceleration $g = 9.81 m^2/s$).
```
Comment (jbathmann) [inline web/content/docs/benchmarks/hydro-component/theis/HC_Theis.pandoc]: Didn't see this in the first round, but 
$\gamma = \frac{S_s}{b \phi g \rho}$
-->
$\gamma = \frac{S_s}{b \phi g \rho_0}$

**Pull request merged**